custom_components/bmw_connected_drive/sensor.py

```diff
@@ -191,14 +188,17 @@ def name(self) -> str:
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        vehicle_state = self._vehicle.state.vehicle_status
-        charging_state = vehicle_state.charging_status in [ChargingState.CHARGING]
 
         if self._attribute == "charging_level_hv":
+            vehicle_state = self._vehicle.state.vehicle_status
+            charging_state = vehicle_state.charging_status in [ChargingState.CHARGING]
             return icon_for_battery_level(
                 battery_level=vehicle_state.charging_level_hv, charging=charging_state
             )
-        icon, _ = self._attribute_info.get(self._attribute, [None, None])
+        elif "Destination_" in self._attribute:
```

Comment:
Is this specific check needed? As in doesn't it work using
```icon, _ = self._attribute_info.get(self._attribute, [None, None])```?

Reply:
not needed anymore, removed